### PR TITLE
Run rustfmt on generated code

### PR DESCRIPTION
**Pull request by EPashkin**
@GuillaumeGomez , @sdroege 

Not fully sure that it good idea: as on windows it runs on gtk for 27sec instead of 0.5sec
and single `cargo fmt` runs for 2 sec,
so I hope that on linux it runs faster.

Also, I am completely against using rustfmt as a library now.

### CODE REVIEW

**Comment by GuillaumeGomez** [comment]
I'm fine with it but it might be "unexpected" for some people, no?

**Comment by EPashkin** [comment]
IMHO if we do this option "opt-in" it will be used by no one

**Comment by sdroege** [comment]
Maybe it's so slow because your serialize everything on formatting and do it file by file? `cargo fmt` probably does everything in parallel.

Might be looking into what `cargo fmt` actually does.

**Comment by EPashkin** [comment]
IMHO it most reason why it so slow is windows (slower spawn)+antivirus,
I have same problem when `git` was called for all files to get version 😉 .

`Cargo fmt` uses rustfmt as library without cost of spawning at minimum.

Can you time this version on gtk with and without "--disable_format" option?

**Comment by sdroege** [comment]
4.4s without parameters, 1.51s with `--disable_format`.

Why don't we call rustfmt as a library if the spawning is so expensive? Or why don't we spawn rustfmt only once for all files at the end, i.e. just call `cargo fmt` in the end?

**Comment by EPashkin** [comment]
I don't like to add big dependency, that also still have "-nightly" suffix.
I like idea about just running  "cargo fmt"

**Comment by EPashkin** [comment]
Use `cargo fmt` now

**Comment by EPashkin** [comment]
```
Timing information for Gir:                                                                                                                     Total - 1 * 5.0s = 5.0s (100.0%)                                                                                                                Loading - 1 * 2.7s = 2.7s (53.7%)                                                                                                             Preprocessing - 1 * 0ns = 0ns (0.0%)                                                                                                          Update library by config - 1 * 0ns = 0ns (0.0%)                                                                                               Postprocessing - 1 * 14.6ms = 14.6ms (0.3%)                                                                                                   Resolving type ids - 1 * 0ns = 0ns (0.0%)                                                                                                     Checking versions - 1 * 0ns = 0ns (0.0%)                                                                                                      Namespace/symbol/class analysis - 1 * 6.9ms = 6.9ms (0.1%)                                                                                    Analyzing - 1 * 150.0ms = 150.0ms (3.0%)                                                                                                      Generating - 1 * 229.8ms = 229.8ms (4.6%)                                                                                                     Formatting - 1 * 1.9s = 1.9s (38.4%)   
```

**Comment by EPashkin** [comment]
So speed problem is "solved", IMHO now only left is to decide: opt-in, opt-out (current) or this feature not needed.

**Comment by sdroege** [comment]
> I don't like to add big dependency, that also still have "-nightly" suffix.

It's using the compiler infrastructure, so always has to be nightly :)

> IMHO now only left is to decide: opt-in, opt-out (current) or this feature not needed.

opt-out seems fine to me

File: src/main.rs [hunk 1]
```diff
@@ -80,6 +81,7 @@ fn build_config() -> Result<Config, String> {
         matches.opt_str("doc-target-path").as_str_ref(),
         matches.opt_present("b"),
         matches.opt_present("s"),
+        matches.opt_present("disable_format"),
```
Comment (sdroege) [inline src/main.rs]:
```suggestion
        matches.opt_present("disable-format"),
```

**Comment by EPashkin** [comment]
Then I will merge when CI pass.

**Comment by sdroege** [comment]
Now we can remove it from the scripts :)

**Pull request merged**